PiwikTracker/Dispatcher.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+internal protocol Dispatcher {
+    associatedtype Dispatchable
+    
+    var userAgent: String? { get set }
+    
+    func send(event: Dispatchable, success: ()->(), failure: (_ shouldContinue: Bool)->());
```

Comment:
I think the Dispatcher protocol is good. I just think we should either constraint the Dispatchable or use Event directly.
You could constraint Dispatchable with a protocol containing one property 'URLRequest' that return the same type. Then let Event implement this protocol.